### PR TITLE
[hotfix](dev-1.0.1) 5 concurrent "insert...select..." OOM

### PR DESCRIPTION
**Pull request by englefly**
# Proposed changes
Five concurrent execution of"insert ... select..."  meet OOM.
With this pr, the peak mem_usage is 44G.(batch_size=1024)
changes:
1. in VOlapScanNode, add limit on block byte size when block merge
2. in NodeChannel, if data are more than 2M, we send it. 
3. set default load_process_max_memory_limit_percent to 50 (origin 80)

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
7. Has unit tests been added: (Yes/No/No Need)
8. Has document been added or modified: (Yes/No/No Need)
9. Does it need to update dependencies: (Yes/No)
10. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/tablet_sink.cpp [hunk 1]
```diff
@@ -315,9 +315,19 @@ Status NodeChannel::add_row(BlockRow& block_row, int64_t tablet_id) {
         SCOPED_ATOMIC_TIMER(&_mem_exceeded_block_ns);
         SleepFor(MonoDelta::FromMilliseconds(10));
     }
-
+    constexpr size_t BATCH_SIZE_FOR_SEND = 2 * 1024 * 1024; //2M
     auto row_no = _cur_batch->add_row();
-    if (row_no == RowBatch::INVALID_ROW_INDEX) {
+    if (row_no == RowBatch::INVALID_ROW_INDEX || _cur_batch->total_byte_size() > BATCH_SIZE_FOR_SEND) {
```
Comment (yiguolei) [inline be/src/exec/tablet_sink.cpp:320]: 使用tuple pool的size 来计算

File: be/src/exec/tablet_sink.h [hunk 2]
```diff
@@ -227,6 +227,11 @@ class NodeChannel {
                            _node_info.brpc_port);
     }
 
+    size_t get_pending_bytes() {
+        size_t v = _pending_batches_bytes;
```
Comment (yiguolei) [inline be/src/exec/tablet_sink.h:231]: 这个不需要这么写

File: be/src/vec/exec/volap_scan_node.cpp [hunk 3]
```diff
@@ -228,8 +228,12 @@ void VOlapScanNode::scanner_thread(VOlapScanner* scanner) {
             std::lock_guard<std::mutex> l(_free_blocks_lock);
             _free_blocks.emplace_back(block);
         } else {
+            // block should not exceed MAX_BLOCK_SIZE,
+            // in order to avoid generating supper huge block when reading big-wide-table
+            constexpr size_t MAX_BLOCK_SIZE = 1024 * 1024 * 100;
             if (!blocks.empty() &&
-                blocks.back()->rows() + block->rows() <= _runtime_state->batch_size()) {
+                blocks.back()->rows() + block->rows() <= _runtime_state->batch_size() &&
```
Comment (yiguolei) [inline be/src/vec/exec/volap_scan_node.cpp:235]: 这个地方可能不用

File: be/src/exec/tablet_sink.cpp (hunk 1 above)
Comment (englefly) [reply be/src/exec/tablet_sink.cpp:320]: done

File: be/src/exec/tablet_sink.h (hunk 2 above)
Comment (englefly) [reply be/src/exec/tablet_sink.h:231]: done

File: be/src/vec/exec/volap_scan_node.cpp (hunk 3 above)
Comment (englefly) [reply be/src/vec/exec/volap_scan_node.cpp:235]: 将外层while循环的条件调整，遇到 raw_bytes_read >= raw_bytes_threshold  就推出循环，这样可以避免大宽表时生成巨大的block

File: be/src/exec/tablet_sink.cpp [hunk 4]
```diff
@@ -315,9 +315,19 @@ Status NodeChannel::add_row(BlockRow& block_row, int64_t tablet_id) {
...
+        //if pending bytes is more than 1G, wait
+        constexpr size_t MAX_PENDING_BYTES = 1073741824; //1G
+        if (_pending_batches_bytes>1073741824){
```
Comment (xinyiZzz) [inline be/src/exec/tablet_sink.cpp:323]: 之后这里用 MemTracker 的值作为判断条件是不是更合适

Comment (englefly) [reply be/src/exec/tablet_sink.cpp:323]: master版本可以

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by morningman — CHANGES_REQUESTED** [review]
Need more test, holding

**Review by morningman — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**